src/audio/aria/aria.c

```diff
@@ -188,21 +188,9 @@ static void aria_free(struct comp_dev *dev)
 static void aria_set_stream_params(struct comp_buffer *buffer, struct aria_data *cd)
 {
 	struct comp_buffer __sparse_cache *buffer_c;
-	enum sof_ipc_frame valid_fmt, frame_fmt;
 
 	buffer_c = buffer_acquire(buffer);
-
-	audio_stream_fmt_conversion(cd->base.audio_fmt.depth,
-				    cd->base.audio_fmt.valid_bit_depth,
-				    &frame_fmt, &valid_fmt,
-				    cd->base.audio_fmt.s_type);
-	audio_stream_set_buffer_fmt(&buffer_c->stream,
-				    cd->base.audio_fmt.interleaving_style);
-	audio_stream_set_frm_fmt(&buffer_c->stream, frame_fmt);
-	audio_stream_set_valid_fmt(&buffer_c->stream, valid_fmt);
-	audio_stream_set_channels(&buffer_c->stream, cd->chan_cnt);
-	audio_stream_set_rate(&buffer_c->stream, cd->base.audio_fmt.sampling_frequency);
-
+	ipc4_update_buffer_format(buffer_c, &cd->base.audio_fmt);
```

Comment:
Nice, I didn't know about this function! It helps to simplify other module adapter conversion patches.